src/dune_pkg/lock_dir.ml

```diff
@@ -339,7 +339,31 @@ let to_dyn { version; packages; ocaml; repos; expanded_solver_variable_bindings
     ]
 ;;
 
+(* [validate_packages packages] returns
+   [Error (`Missing_dependency (dep, pkg, loc))] if [packages] doesn't have a
+   key [dep], but [dep] is a dependency of package [pkg] at location [loc] *)
+let validate_packages packages =
+  Package_name.Map.values packages
+  |> List.find_map ~f:(fun (pkg : Pkg.t) ->
+    List.find_map pkg.deps ~f:(fun (loc, dep) ->
+      if Package_name.Map.mem packages dep
+      then None
+      else Some (`Missing_dependency (dep, pkg, loc))))
```

Comment:
How about reporting all the missing dependencies?

Reply:
updated